region/maridia/inner-pink/The Precious Room.json

```diff
@@ -494,6 +494,57 @@
         "canCrossRoomJumpIntoWater"
       ]
     },
+    {
+      "link": [2, 1],
+      "name": "Use Flash Suit",
+      "requires": [
+        {"useFlashSuit": {}},
+        {"or": [
+          {"shinespark": {"frames": 37, "excessFrames": 9}},
+          {"and": [
+            "HiJump",
+            {"shinespark": {"frames": 33, "excessFrames": 8}}
+          ]}
+        ]}
+      ],
+      "flashSuitChecked": true
+    },
+    {
+      "link": [2, 1],
+      "name": "Cross Room Jump, Use Flash Suit",
+      "entranceCondition": {
+        "comeInJumping": {
+          "speedBooster": "any",
+          "minTiles": 0.4375
+        }
+      },
+      "requires": [
+        "canCrossRoomJumpIntoWater",
+        {"useFlashSuit": {}},
+        {"shinespark": {"frames": 16, "excessFrames": 9}}
+      ],
+      "flashSuitChecked": true,
+      "note": [
+        "Cross room jump onto the top middle platform before sparking.",
+        "Requires a runway of 1 tile in the adjacent room."
+      ],
+      "devNote": [
+        "Assumes Samus lands, then jumps again before sparking.",
+        "There is no HiJump variant, as that can jump out of the water."
+      ]
+    },
+    {
+      "link": [2, 1],
+      "name": "Long Underwater Wall Jump, Use Flash Suit",
+      "requires": [
+        "canTrickyCarryFlashSuit",
+        "canUnderwaterWalljump",
+        {"useFlashSuit": {}},
+        {"shinespark": {"frames": 12, "excessFrames": 8}}
+      ],
+      "flashSuitChecked": true,
+      "note": "Underwater wall jump to the top middle platform before sparking to save Energy."
```

Comment:
About as hard as notable extreme uwjs.  The risk of using the flash suit is in the case where you break spin and fail the uwj.  Interestingly map rando can assume usage of spin lock, but here we assume using controllers and no spin lock.  Fine for base insane.

Reply:
I was thinking that this being in insane would be ok, but do you think it should also be notable? I could see players wanting all long uwu to be toggleable

Reply:
I generally can't do the notable uwjs that are in extreme without spinlock, but would be fine doing tricky flash suit movements.  So the ability to toggle off this strat makes sense.  
Since there is a set of uwjs that are similarly hard, we could look at making them a tech.  I would look at moving current uwjs down to canShortUnderwaterWalljump, since you only need to do it a few times to gain enough height.